californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -364,6 +365,15 @@ public void run() {
 					LOGGER.log(Level.FINE, "Timeout: retransmission limit reached, exchange failed, message: {0}", message);
 					exchange.setTimedOut();
 					message.setTimedOut(true);
+					OptionSet options = message.getOptions();
+					boolean isBlockwise = options.hasBlock1() || options.hasBlock2();
+					if (message instanceof Request && isBlockwise) {
+					    Request request = exchange.getRequest();
+					    if (request != null && request.getMID() != message.getMID()) {
+					        // Notifying the original request about timeout
+					        request.setTimedOut(true);
+					    }
```

Comment:
Yes, and it is actually not that easy to get a grip on such an intermediary request from user code (usually these intermediary reuqests are created and issued by Californium). However, a seasoned user can also do blockwise transfer *manually*, i.e. using a sequence of manually crafted and sent GET requests to perform the transfer.